app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaViewModel.kt

```diff
@@ -0,0 +1,382 @@
+import com.wire.android.ui.home.conversationslist.model.ConversationInfo
+import com.wire.android.ui.home.conversationslist.model.ConversationItem
+import com.wire.android.ui.home.conversationslist.parseConversationEventType
+import com.wire.android.ui.home.conversationslist.parsePrivateConversationEventType
+import com.wire.android.ui.home.conversationslist.showLegalHoldIndicator
+import com.wire.android.util.FileManager
+import com.wire.android.util.ImageUtil
+import com.wire.android.util.dispatchers.DispatcherProvider
+import com.wire.android.util.getMetaDataFromUri
+import com.wire.android.util.getMimeType
+import com.wire.android.util.isImageFile
+import com.wire.android.util.parcelableArrayList
+import com.wire.android.util.resampleImageAndCopyToTempPath
+import com.wire.android.util.ui.WireSessionImageLoader
+import com.wire.kalium.logic.data.asset.KaliumFileSystem
+import com.wire.kalium.logic.data.conversation.ConversationDetails
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.feature.asset.GetAssetSizeLimitUseCase
+import com.wire.kalium.logic.feature.asset.ScheduleNewAssetMessageUseCase
+import com.wire.kalium.logic.feature.conversation.ObserveConversationListDetailsUseCase
+import com.wire.kalium.logic.feature.user.GetSelfUserUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.joinAll
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.util.UUID
+import javax.inject.Inject
+
+@HiltViewModel
+@Suppress("LongParameterList", "TooManyFunctions")
+class ImportMediaViewModel @Inject constructor(
+    private val getSelf: GetSelfUserUseCase,
+    private val userTypeMapper: UserTypeMapper,
+    private val observeConversationListDetails: ObserveConversationListDetailsUseCase,
+    private val fileManager: FileManager,
+    private val navigationManager: NavigationManager,
+    private val sendAssetMessage: ScheduleNewAssetMessageUseCase,
+    private val kaliumFileSystem: KaliumFileSystem,
+    private val getAssetSizeLimit: GetAssetSizeLimitUseCase,
+    val wireSessionImageLoader: WireSessionImageLoader,
+    val dispatchers: DispatcherProvider,
+) : ViewModel() {
+    var importMediaState by mutableStateOf(
+        ImportMediaState()
+    )
+        private set
+
+    var shareableConversationListState by mutableStateOf(ShareableConversationListState())
+        private set
+
+    val selectedConversationFlow = MutableStateFlow(emptyList<ConversationItem>())
+
```

Comment:
this is not how MVI is supposed to expose a state, use a single data class for a single state for this ViewModel instead

Reply:
done